src/esm_parser/esm_parser.py

```diff
@@ -1468,9 +1468,19 @@ def resolve_basic_choose(config, config_to_replace_in, choose_key, blackdict={})
         choice = do_math_in_entry([False], choice, config)
     logging.debug(choice)
 
-    if choice in config_to_replace_in.get(choose_key):
+    # This allows users to use version numbers (floats) and integers as choices inside
+    # the choose_blocks, instead of having to specify the choices as strings
+    choices_available = {}
+    for ckey, cval in config_to_replace_in.get(choose_key, {}).items():
```

Comment:
I am afraid that this part can cause some issues. `items()` method works on dictionaries. What happens if `config_to_replace_in['choose_key']` is not a dictionary?

Maybe it would be a better idea to place an `isinstance` check like
`if isinstance(config_to_replace_in.get(choose_key, {}), dict)`
It looks like the previous code did not crash so far since it has 
`six.iteritems(config_to_replace_in[choose_key][choice])` on line 1473. So `config_to_replace_in[choose_key][choice]` yields a dictionary so far.

Reply:
You are right. But bare in mind that the contents of a `choose_` block need to always be a dictionary:

```
choose_<var>:
        <choice_1>: <var/dict/list>
        <choice_2>: <var/dict/list>
```

What do you think about placing a check here to return a user-friendly error if a `choose` is not a dictionary instead?

```
 if not isinstance(config_to_replace_in.get(choose_key, {}), dict):
    user_error("choose_ block", "``choose_`` blocks need to be defined as ``dictionaries``. And some more text that tells the user what is going on and what exact choose is giving problems")
```

Reply:
Yes, this is a good solution. Good error messages prevent headaches later on 😄

Reply:
I'll do that then.

Reply:
Yes to good error messages. 👍🏻

Reply:
...but, I would still like to find a way to force strings for version numbers. It is (in my opinion) cleaner. What if you get something like:

```
echam v1.6.04p2rc-alpha2
```

I have never seen that "in the wild" of course, but some elements should just be strings, always. I would count version info to that list.

Reply:
But that should go somewhere else right? I mean, this is specific for choices in `choose_` blocks. While what you are proposing is more of a syntax check/transformation that should go for example in the `yaml_to_dict` method, right?

Reply:
@pgierz , I think that `echam v1.6.04p2rc-alpha2` will be interpreted by the yaml parser as a `str` anyway, so it will stay being a string